ui/main/src/app/modules/share/light-card/light-card.component.ts

```diff
@@ -113,11 +122,25 @@ export class LightCardComponent implements OnInit, OnDestroy {
         }
     }
 
+    private computeGroupedCardsIcon() {
+        this.showGroupedCardsIcon = this.groupedCardsService.isParentGroupCard(this.lightCard);
+    }
+
+    getGroupedChildCards() {
+        return this.groupedCardsService.getChildCardsByTags(this.lightCard.tags);
+    }
+
     handleDate(timeStamp: number): string {
         return this.time.formatDateTime(timeStamp);
     }
 
-    public select() {
+    public select($event) {
+        $event.stopPropagation();
+        if (this.open && this.groupedCardsService.isParentGroupCard(this.lightCard)) {
+            this.groupedCardsVisible = ! this.groupedCardsVisible;
+        } else {
+            this.groupedCardsVisible = true;
+        }
```

Comment:
to be call only if group option is activated to avoid unnecessary processing

Reply:
Same as above.